assets/js/components/DateRangeSelector.js

```diff
@@ -19,40 +19,93 @@
 /**
  * WordPress dependencies
  */
-import { useCallback } from '@wordpress/element';
+import { useCallback, useEffect, useRef, useState } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import Data from 'googlesitekit-data';
-import { Option, Select } from '../material-components';
+import DateRangeIcon from '../../svg/date-range.svg';
+import Menu from './menu';
 import { getAvailableDateRanges } from '../util/date-range';
 import { STORE_NAME as CORE_USER } from '../googlesitekit/datastore/user/constants';
+import Button from './button';
+
 const { useSelect, useDispatch } = Data;
 
 function DateRangeSelector() {
 	const ranges = Object.values( getAvailableDateRanges() );
 	const dateRange = useSelect( ( select ) => select( CORE_USER ).getDateRange() );
 	const { setDateRange } = useDispatch( CORE_USER );
-	const onChange = useCallback( ( index, item ) => {
-		setDateRange( item.dataset.value );
-	}, [ ranges ] );
+
+	const [ menuOpen, setMenuOpen ] = useState( false );
+	const menuButtonRef = useRef();
+	const menuRef = useRef();
+
+	useEffect( () => {
+		const handleMenuClose = ( event ) => {
+			// Close the menu if the user presses the Escape key
+			// or if they click outside of the menu.
+			if (
+				( ( 'keyup' === event.type && 27 === event.keyCode ) || 'mouseup' === event.type ) &&
+				! menuButtonRef.current.buttonRef.current.contains( event.target ) &&
+				! menuRef.current.menuRef.current.contains( event.target )
+			) {
+				setMenuOpen( false );
+			}
+		};
+
+		global.addEventListener( 'mouseup', handleMenuClose );
+		global.addEventListener( 'keyup', handleMenuClose );
+
+		return () => {
+			global.removeEventListener( 'mouseup', handleMenuClose );
+			global.removeEventListener( 'keyup', handleMenuClose );
+		};
+	}, [] );
+
+	const handleMenu = useCallback( () => {
+		setMenuOpen( ! menuOpen );
+	}, [ menuOpen ] );
+
+	const handleMenuItemSelect = useCallback( ( index, event ) => {
+		if (
+			( 'keydown' === event.type && ( 13 === event.keyCode || 32 === event.keyCode ) ) || // Enter or Space is pressed.
+			'click' === event.type // Mouse is clicked
+		) {
+			setDateRange( ranges[ index ].slug );
+			setMenuOpen( false );
+		}
+	}, [ handleMenu ] );
+
+	const currentDateRangeLabel = ranges.reduce( ( acc, range ) => {
+		return range.slug === dateRange ? range : acc;
+	}, {} ).label;
+
+	const menuItems = ranges.map( ( range ) => range.label );
 
 	return (
-		<Select
-			enhanced
-			className="mdc-select--minimal"
-			name="time_period"
-			label=""
-			onEnhancedChange={ onChange }
-			value={ dateRange }
-		>
-			{ ranges.map( ( { slug, label } ) => (
-				<Option key={ slug } value={ slug }>
-					{ label }
-				</Option>
-			) ) }
-		</Select>
+		<div className="googlesitekit-date-range-selector">
```

Comment:
Good shout. I've managed to remove a good few lines after working through each class and seeing what's needed. I still need to reference `.googlesitekit-dropdown-menu` in the file because this ticket included reducing the font-weight on the dropdown label and increasing the image size, both of which I have to overwrite here.